cmdstanpy/utils.py

```diff
@@ -128,46 +128,127 @@ def check_csv(filename: str) -> Dict:
 def scan_stan_csv(filename: str) -> Dict:
     """Process stan_csv file line by line."""
     dict = {}
-    draws_found = 0
     lineno = 0
-    with open(filename) as fp:
-        line = fp.readline().strip()
+    try:
+        fp = open(filename, 'r')
+        try:
+            lineno = scan_config(fp, dict, lineno)
+            lineno = scan_column_names(fp, dict, lineno)
+            lineno = scan_metric(fp, dict, lineno)
+            lineno = scan_draws(fp, dict, lineno)
+        finally:
+            fp.close()
+    except IOError:
+        raise IOError()
+    return dict
+
+
+def scan_config(fp: TextIO, dict: Dict, lineno: int) -> int:
+    """
+    Scan initial stan_csv file comments lines and
+    save non-default configuration information to dict.
+    """
+    cur_pos = fp.tell()
+    line = fp.readline().strip()
+    while len(line) > 0 and line.startswith('#'):
         lineno += 1
-        while len(line) > 0 and line.startswith('#'):
-            parse_header(line, dict)
-            line = fp.readline().strip()
-            lineno += 1
-        dict['column_names'] = tuple(line.split(','))
-        num_cols = len(dict['column_names'])
+        if not line.endswith('(Default)'):
+            line = line.lstrip(' #\t')
+            key_val = line.split('=')
+            if len(key_val) == 2:
+                if key_val[0].strip() == 'file' and not key_val[1].endswith(
+                        'csv'):
+                    dict['data_file'] = key_val[1].strip()
+                elif key_val[0].strip() != 'file':
+                    dict[key_val[0].strip()] = key_val[1].strip()
+        cur_pos = fp.tell()
         line = fp.readline().strip()
-        lineno += 1
-        while len(line) > 0 and line.startswith('#'):
-            line = fp.readline().strip()
+    fp.seek(cur_pos)
```

Comment:
I don't remember if there are some problems on Windows with tell and seek.

Reply:
web search "python" "seek" "windows" only turned up old bug in python 2 about seek and write.